packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -0,0 +1,93 @@
+import { Aliases, AuthInfo } from '@salesforce/core';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import { ForceConfigGet } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+
+export async function setupWorkspaceOrgType() {
+  const defaultUsernameOrAlias = await getDefaultUsernameOrAlias();
+  const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+  let isScratchOrg = false;
+  if (defaultUsernameIsSet) {
+    const username = await getUsername(defaultUsernameOrAlias!);
+    try {
+      isScratchOrg = await isAScratchOrg(username);
+    } catch (e) {
+      if (e.name === 'NamedOrgNotFound') {
+        // TODO: what should we do in this case? Expose all source commands by default?
+        setDefaultOrgIsScratchOrg(true);
+        setDefaultOrgIsNonScratchOrg(true);
+        return;
+      } else {
+        throw e;
+      }
+    }
+  }
+  setDefaultOrgIsScratchOrg(defaultUsernameIsSet && isScratchOrg);
+  setDefaultOrgIsNonScratchOrg(defaultUsernameIsSet && !isScratchOrg);
+}
+
+async function isAScratchOrg(username: string): Promise<boolean> {
+  const authInfo = await AuthInfo.create(username);
+  const authInfoFields = authInfo.getFields();
+  return Promise.resolve(typeof authInfoFields.devHubUsername !== 'undefined');
+}
+
+/**
+ * Returns the non-aliased username
+ * @param usernameOrAlias
+ */
+async function getUsername(usernameOrAlias: string): Promise<string> {
+  const username = await Aliases.fetch(usernameOrAlias);
+  if (username) {
+    return Promise.resolve(username);
+  }
+  return Promise.resolve(usernameOrAlias);
+}
+
+function setDefaultOrgIsScratchOrg(val: boolean) {
+  vscode.commands.executeCommand(
+    'setContext',
+    'sfdx:default_org_is_scratch_org',
+    val
+  );
+}
+
+function setDefaultOrgIsNonScratchOrg(val: boolean) {
+  vscode.commands.executeCommand(
+    'setContext',
+    'sfdx:default_org_is_non_scratch_org',
+    val
+  );
+}
+
+async function getDefaultUsernameOrAlias(): Promise<string | undefined> {
+  if (
+    vscode.workspace.workspaceFolders instanceof Array &&
+    vscode.workspace.workspaceFolders.length > 0
+  ) {
+    const forceConfig = await new ForceConfigGet().getConfig(
+      vscode.workspace.workspaceFolders[0].uri.fsPath,
+      'defaultusername'
+    );
+    return forceConfig.get('defaultusername');
+  }
+}
+
+export function registerDefaultOrgWatcher(context: vscode.ExtensionContext) {
+  if (
+    vscode.workspace.workspaceFolders instanceof Array &&
+    vscode.workspace.workspaceFolders.length > 0
+  ) {
+    const sfdxConfigWatcher = vscode.workspace.createFileSystemWatcher(
```

Comment:
What happens if this file is not available yet?
AFAIK .sfdx/sfdx-config.json is only available after you have executed one of the auth commands.
Prior to that it doesn't exist.

Can we register a watcher first (when the files don't exist) and when the files are created, the watcher is smart enough to register?

Reply:
Yea, the onDidCreate() listener picks up on create events and I manually verified that it works

Reply:
Your suggestion to test on windows was a good call. I was previously using `vscode.RelativePattern()` to create the pattern passed to `vscode.workspace.createFileSystemWatcher` and that was not working on Windows. The new implementation works on Mac and Windows.